### PR TITLE
fix: web config tool including \r in prompt

### LINKED ISSUE #4960: Web config tool includes \r in prompt
When using the web config (xonfig web), newlines are interpreted as "\r\n". On macOS (and Linux, I assume) this adds a "^M" to the prompt, before the newline. 
## xonfig
```
+------------------+--------------------------------+
| xonsh            | 0.13.3                         |
| Python           | 3.10.7                         |
| PLY              | 3.11                           |
| have readline    | True                           |
| prompt toolkit   | 3.0.31                         |
| shell type       | prompt_toolkit                 |
| history backend  | json                           |
| pygments         | 2.13.0                         |
| on posix         | True                           |
| on linux         | False                          |
| on darwin        | True                           |
| on windows       | False                          |
| on cygwin        | False                          |
| on msys2         | False                          |
| is superuser     | False                          |
| default encoding | utf-8                          |
| xonsh encoding   | utf-8                          |
| encoding errors  | surrogateescape                |
| xontrib          | []                             |
+------------------+--------------------------------+
```

## Expected Behavior
I would expect the "\r" to be stripped from the text field before the prompt is set. 

## Current Behavior
I was able to confirm that the "\r" is being added to the prompt variable by checking my .xonshrc after using the web config. Removing this value removes the "^M" from my prompt.

## Steps to Reproduce
* Run xonfig web from your shell.
* Select a multi-line prompt, and set your .xonshrc
* Exit and restart xonsh. You'll see a "^M" before the new line.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### PR DESCRIPTION
**Pull request by peipacut**
hi guys, this is my first commit. please let me know if i've done anything wrong :)

previously using the xonsh web config tool to set a prompt with a newline added "^M" at the end of the prompt. this fix strips all \r characters in the config file and strips all \r characters in the prompt page's post request.

closes #4960 

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by jnoortheen** [comment]
pre-commit.ci autofix

File: xonsh/webconfig/routes.py [hunk 1]
```diff
@@ -239,10 +240,13 @@ def get(self):
...
     def post(self, data: "cgi.FieldStorage"):
         if data:
-            prompt = data.getvalue(self.var_name)
+            prompt = data.getvalue(self.var_name).replace("\r", "")
             self.env[self.var_name] = prompt
             self.update_rc(prompt=prompt)
 
+    def ee(self, str):
+        return re.sub(r"\\r", "", str)
+
```
Comment (gforsyth) [inline xonsh/webconfig/routes.py:249]: Is this being used or is this left over from debugging?

File: xonsh/webconfig/file_writes.py [hunk 2]
```diff
@@ -39,7 +40,7 @@ def config_to_xonsh(
     for func in RENDERERS:
         lines.extend(func(config))
     lines.append(suffix)
-    return "\n".join(lines)
+    return re.sub(r"\\r", "", "\n".join(lines))
```
Comment (gforsyth) [inline xonsh/webconfig/file_writes.py:43]: Does a `replace` on the joined string work here?  Not a blocker, just curious

**Review by gforsyth — COMMENTED** [review]
Hey @peipacut !  Thanks for putting this in!  Sorry for the delay in looking at it.

A few small questions inline, then I think we can merge this in.

File: xonsh/webconfig/file_writes.py (hunk 2 above)
Comment (peipacut) [reply xonsh/webconfig/file_writes.py:43]: sorry for the late reply. when testing i found it does work and the prompt in the config file is stripped of the carriage return.

File: xonsh/webconfig/routes.py (hunk 1 above)
Comment (peipacut) [reply xonsh/webconfig/routes.py]: removed

**Comment by gforsyth** [comment]
pre-commit.ci autofix

**Review by gforsyth — APPROVED** [review]
Thanks for putting this in @peipacut !

**Pull request merged**